google/cloud/bigquery/job.py

```diff
@@ -3337,7 +3339,49 @@ def to_arrow(
 
         ..versionadded:: 1.17.0
         """
-        return self.result().to_arrow(
+        if self.query_plan and progress_bar_type:
```

Comment:
Is this logic necessary if we are calling `to_arrow`? Can't we rely on `to_arrow`'s progress bar support?

Reply:
Sorry, didn't get this point, are you talking about table.to_arrow() progress bar support?

Reply:
Oh, I was getting this confused with `RowIterator`, where `to_dataframe` relies on `to_arrow`'s progress bar support. I do wonder if some of this "waiting for the query to finish" logic could be refactored into a `_tqdm_helpers.py` function, since it should be identical between `to_dataframe` and `to_arrow`